src/components/wallets/TopupWalletDialog.tsx

```diff
@@ -82,6 +84,7 @@ export const TopupWalletDialog = forwardRef<DialogRef, TopupWalletDialogProps>(
               walletId: wallet.id,
               grantedCredits: grantedCredits === '' ? '0' : String(grantedCredits),
               paidCredits: paidCredits === '' ? '0' : String(paidCredits),
+              invoiceRequiresSuccessfulPayment: formikProps.values.invoiceRequiresSuccessfulPayment,
```

Comment:
Why not getting it from the props but directly from formik object?

Reply:
Nice catch!! Fixing it